internal/controller/promotionstrategy_controller.go

```diff
@@ -316,19 +330,30 @@ func (r *PromotionStrategyReconciler) createOrUpdatePreviousEnvironmentCommitSta
 		},
 	}
 	updatedCS := &promoterv1alpha1.CommitStatus{}
-	err := r.Get(ctx, proposedCSObjectKey, updatedCS)
+	err = r.Get(ctx, proposedCSObjectKey, updatedCS)
 	if err != nil {
 		if errors.IsNotFound(err) {
 			err = r.Create(ctx, commitStatus)
 			if err != nil {
 				return fmt.Errorf("failed to create previous environments CommitStatus: %w", err)
 			}
+			return nil
 		}
-	} else if updatedCS.Spec.Phase != phase || updatedCS.Spec.Sha != ctp.Status.Proposed.Hydrated.Sha {
+		return fmt.Errorf("failed to get previous environments CommitStatus: %w", err)
+	}
+
+	updatedYamlStatusMap := make(map[string]string)
+	err = yaml.Unmarshal([]byte(updatedCS.Annotations[promoterv1alpha1.CommitStatusDeAggregationAnnotation]), &updatedYamlStatusMap)
```

Comment:
Is it safe to assume that this annotation is present?

Reply:
Yes we create this just above so it should always exist

Reply:
added a check still